src/modules/commander/Commander.cpp

```diff
@@ -1237,6 +1237,31 @@ Commander::set_home_position_alt_only()
 	return false;
 }
 
+void
+Commander::updateHomePositionYaw(float yaw)
+{
+	home_position_s home = _home_pub.get();
+
+	home.yaw = yaw;
+	home.timestamp = hrt_absolute_time();
+
+	_home_pub.update(home);
+}
+
+void
+Commander::checkEkfResetCounters()
+{
+	if (_attitude_sub.get().quat_reset_counter != _quat_reset_counter) {
+		const float delta_psi = matrix::Eulerf(matrix::Quatf(_attitude_sub.get().delta_q_reset)).psi();
+
+		if (!_home_pub.get().manual_home) {
+			updateHomePositionYaw(matrix::wrap_pi(_home_pub.get().yaw + delta_psi));
+		}
+
+		_quat_reset_counter = _attitude_sub.get().quat_reset_counter;
```

Comment:
Estimator side is there any limit to the reset frequency? Commander is running at a lower rate and lower priority than the estimator. Depending on the system specifics and configuration it's only going to be seeing every 1/2 - 1/4 `vehicle_attitude` updates.

Reply:
IIRC, there is only a limitation to avoid resetting several times per frame, but that's all. However, I've never seen it resetting more than once every several seconds; this is mainly due to the fact that after the reset, the yaw needs a few seconds to diverge again.
In the usual use case, there is only a reset at boot, then one at 1.5m. If there are issues, we can have one or two more but then the yaw estimator takes over and there is no further reset.